src/mem.rs

```diff
@@ -100,6 +100,7 @@ impl MmapMemory {
 	/// This is unsafe, as can create multiple aliasing. During the lifetime of
 	/// the returned slice, the memory must not be altered to prevent undfined
 	/// behaviour.
+	#[allow(clippy::mut_from_ref)]
```

Comment:
this shouldn't be necessary (because it returns an immutable ref)

Reply:
```
error: mutable borrow from immutable input(s)
  --> src/mem.rs:81:39
   |
81 |     pub unsafe fn as_slice_mut(&self) -> &mut [u8] {
   |                                          ^^^^^^^^^
   |
note: immutable borrow here
  --> src/mem.rs:81:29
   |
81 |     pub unsafe fn as_slice_mut(&self) -> &mut [u8] {
   |                                ^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mut_from_ref
   = note: `#[deny(clippy::mut_from_ref)]` on by default
```

Reply:
I'm going to get another coffee.

Reply:
TODO: Use expect instead of allow (so as to get warned for unnecessary lints)